packages/types/src/dedupeCheckIssues.ts

```diff
@@ -0,0 +1,30 @@
+export interface DedupeCheckIssues {
+  readonly importerIssuesByImporterId: SnapshotsChanges
+  readonly packageIssuesByDepPath: SnapshotsChanges
+}
+
+export interface SnapshotsChanges {
+  readonly added: readonly string[]
+  readonly removed: readonly string[]
+  readonly updated: Record<string, ResolutionChangesByAlias>
+}
+
+export type ResolutionChangesByAlias = Record<string, ResolutionChange>
+
+export type ResolutionChange = ResolutionAdded | ResolutionDeleted | ResolutionUpdated
+
+export interface ResolutionAdded {
+  readonly type: 'added'
+  readonly next: string
+}
+
+export interface ResolutionDeleted {
+  readonly type: 'removed'
+  readonly prev: string
+}
+
+export interface ResolutionUpdated {
+  readonly type: 'updated'
```

Comment:
it might be better to not use a keyword for the field name.

what about naming it `change`?

Reply:
but it might be even not needed because it is clear without this additional field what was the change.

Reply:
> it might be better to not use a keyword for the field name.

Were we thinking an enum instead? Something like the `ShapeKind` mentioned in the TypeScript docs? https://www.typescriptlang.org/docs/handbook/enums.html#union-enums-and-enum-member-types

```ts
enum ShapeKind {
  Circle,
  Square,
}
```

> what about naming it `change`?

Would we rename `ResolutionUpdated` to `ResolutionChange` as well? We would need a new name for the top level tagged union.

```ts
export type ResolutionChange = ResolutionAdded | ResolutionDeleted | ResolutionUpdated
```

I could imagine the current `ResolutionChange` becoming `ResolutionDiff`.

> but it might be even not needed because it is clear without this additional field what was the change.

I do like the type safety of the [tagged union](https://www.typescriptlang.org/docs/handbook/enums.html#union-enums-and-enum-member-types). I think it provides a better guarantee at compile time that we don't mistaken the `ResolutionUpdated` objects for `ResolutionDeleted` or `ResolutionAdded`. We could check `obj.prev != null && obj.next != null`, but that introduces possibility for mistakes both when reading the object and creating it.